src/Microsoft.AspNetCore.Razor.Tools/ServerProtocol/ServerConnection.cs

```diff
@@ -290,7 +290,7 @@ internal static bool TryCreateServerCore(string clientDir, string pipeName, out
             // The server should be in the same directory as the client
             var expectedCompilerPath = Path.Combine(clientDir, ServerName);
             expectedPath = Environment.GetEnvironmentVariable("DOTNET_HOST_PATH") ?? "dotnet";
-            processArguments = $@"""{expectedCompilerPath}"" {(debug ? "--debug" : "")} server -p {pipeName}";
+            processArguments = $@"""{expectedCompilerPath}"" {(debug ? "--debug" : "")} server -p ""{pipeName}""";
```

Comment:
😢

Reply:
Are there other places we manually start a process? Can you double check?

Reply:
Should we use https://github.com/aspnet/Common/blob/dev/shared/Microsoft.Extensions.CommandLineUtils.Sources/Utilities/ArgumentEscaper.cs#L24?

Reply:
> Can you double check?

Just did. There is just this and tests.

> Should we use https://github.com/aspnet/Common/blob/dev/shared/Microsoft.Extensions.CommandLineUtils.Sources/Utilities/ArgumentEscaper.cs#L24?

I'm not sure actually. I did a quick check and looks like we only use it mostly in BuildTools. Was it intended for using in product code?

Reply:
@natemcmaster ?